aspnetcore/fundamentals/configuration.md

```diff
@@ -5,162 +5,981 @@ ms.author: riande
+<!-- literal_block {"xml:space": "preserve", "language": "c#", "dupnames": [], "linenos": false, "classes": [], "ids": [], "backrefs": [], "source": "/Users/shirhatti/src/Docs/aspnet/fundamentals/configuration/sample/src/UsingOptions/Startup.cs", "highlight_args": {"linenostart": 1, "hl_lines": [5, 6, 7, 8, 10, 13, 15, 17, 18, 20, 21]}, "names": []} -->
 
-It's recommended that you create well-factored settings objects that correspond to certain features within your application, thus following the [Interface Segregation Principle (ISP)](http://deviq.com/interface-segregation-principle/) (classes depend only on the configuration settings they use) as well as [Separation of Concerns](http://deviq.com/separation-of-concerns/) (settings for disparate parts of your app are managed separately, and thus are less likely to negatively impact one another).
+````c#
 
-A simple `MyOptions` class is shown here:
+   public class Startup
+   {
+// RM
+       public Startup(IHostingEnvironment env)
+       {
+           // Set up configuration sources.
+           var builder = new ConfigurationBuilder()
+               .SetBasePath(env.ContentRootPath)
+               .AddJsonFile("appsettings.json", optional: true, reloadOnChange: true);
 
-[!code-csharp[Main](configuration/sample/src/UsingOptions/Models/MyOptions.cs?range=3-7)]
+           Configuration = builder.Build();
+       }
 
-Options can be injected into your application using the `IOptions<TOptions>` accessor service. For example, the following [controller](../mvc/controllers/index.md)  uses `IOptions<MyOptions>` to access the settings it needs to render the `Index` view:
+       public IConfigurationRoot Configuration { get; set; }
 
-[!code-csharp[Main](../fundamentals/configuration/sample/src/UsingOptions/Controllers/HomeController.cs?highlight=3,5,8&range=9-20)]
+       public void ConfigureServices(IServiceCollection services)
+       {
+           // Adds services required for using options.
+           services.AddOptions();
 
->[!TIP]
-> Learn more about [Dependency Injection](dependency-injection.md).
+           // Register the ConfigurationBuilder instance which MyOptions binds against.
+           services.Configure<MyOptions>(Configuration);
 
-To setup the `IOptions<TOptions>` service you call the `AddOptions` extension method during startup in your `ConfigureServices` method:
+           // Add framework services.
+           services.AddMvc();
+       }
 
-[!code-csharp[Main](../fundamentals/configuration/sample/src/UsingOptions/Startup.cs?highlight=4&range=26-30)]
+   ````
 
-<a name=options-example></a>
+The following [controller](../mvc/controllers/index.md)  uses [Dependency Injection](dependency-injection.md) on `IOptions<TOptions>` to access settings:
```

Comment:
Is this sample code supposed to target 1.1? If yes I would recommend using `IOptionsShapshot<T>` in the controller constructor since that supports get new values in new requests after configuration files are modified. It also has minimal overhead if you don't care about changes.